src/components/search/services/PmSeriesApiService.ts

```diff
@@ -0,0 +1,189 @@
+import { BackendSrv } from '@grafana/runtime';
+import { DataSourceInstanceSettings } from '@grafana/data';
+import {
+    SeriesMaybeResponse,
+    SeriesDescQueryParams,
+    SeriesDescResponse,
+    SeriesDescMaybeResponse,
+    SeriesNoRecordResponse,
+    SeriesQueryQueryParams,
+    SeriesQueryResponse,
+    SeriesQueryMaybeResponse,
+    SeriesMetricsQueryParams,
+    SeriesMetricsResponse,
+    SeriesMetricsMaybeResponse,
+    SeriesLabelsQueryParams,
+    SeriesLabelsResponse,
+    SeriesLabelsMaybeResponse,
+} from '../models/endpoints/series';
+import { timeout } from '../utils/utils';
+import Config from '../config/config';
+
+class PmSeriesApiService {
+    private static requestId = 0;
+    baseUrl: string;
+    backendSrv: BackendSrv;
+    headers = {
+        'Content-Type': 'application/json',
+    };
+
+    constructor(instanceSettings: DataSourceInstanceSettings, backendSrv: BackendSrv) {
+        if (!instanceSettings.url) {
+            throw new Error();
+        }
+        this.baseUrl = instanceSettings.url;
+        this.headers = {
+            ...this.headers,
+            ...(instanceSettings.basicAuth ? { Authorization: instanceSettings.basicAuth } : {}),
+        };
+        this.backendSrv = backendSrv;
+    }
+
+    getRequestId() {
+        return (PmSeriesApiService.requestId++).toString();
+    }
+
+    static isNoRecordResponse(response: SeriesMaybeResponse) {
+        if (
+            typeof response === 'object' &&
+            (response as { [key: string]: any }).success !== undefined &&
+            Object.keys(response).length === 1
+        ) {
+            return true;
+        }
+        return false;
+    }
+
+    async descs(params: SeriesDescQueryParams): Promise<SeriesDescResponse> {
+        const { baseUrl, getRequestId, headers, backendSrv } = this;
+        const getParams = new URLSearchParams();
+        getParams.append('series', params.series.join(','));
+        if (getParams.get('series')?.length === 0) {
+            return [];
+        }
+        if (params.client !== undefined) {
+            getParams.append('client', params.toString());
+        }
+        const options = {
+            url: `${baseUrl}/series/descs?${getParams.toString()}`,
+            methods: 'GET',
+            showSuccessAlert: false,
+            requestId: getRequestId(),
```

Comment:
is the `requestId` actually used somewhere?

Reply:
From docs:
```js
    /**
     * Provided by the initiator to identify a particular remote call. An example
     * of this is when a datasource plugin triggers a query. If the request id already
     * exist the backendSrv will try to cancel and replace the previous call with the
     * new one.
     */
```
Probably wont be needed, will remove.

Reply:
Addressed in ba23ab4